synapse/storage/__init__.py

```diff
@@ -58,6 +61,22 @@
 LAST_SEEN_GRANULARITY = 120*1000
 
 
+def get_datastore(hs):
```

Comment:
This might want to be called something like ``new_datastore`` to make it clearer that it makes a datastore.
Also could do with some documentation to explain why it is a thing. It's not immediately obvious why this doesn't happen in the constructor.